src/object.d

```diff
@@ -2104,17 +2104,23 @@ auto byKeyValue(T : V[K], K, V)(T aa) pure nothrow @nogc
                 private void* keyp;
                 private void* valp;
 
-                @property ref key() inout { return *cast(substInout!K*)keyp; }
-                @property ref value() inout { return *cast(substInout!V*)valp; }
+                @property auto key() inout
+                {
+                    return () @trusted { return *cast(substInout!K*)keyp; } ();
+                };
+                @property auto value() inout
+                {
+                    return () @trusted { return *cast(substInout!V*)valp; } ();
+                };
             }
             return Pair(_aaRangeFrontKey(r),
                         _aaRangeFrontValue(r));
         }
-        void popFront() { _aaRangePopFront(r); }
+        void popFront() { return () @trusted { return _aaRangePopFront(r); } (); }
         @property Result save() { return this; }
     }
 
-    return Result(_aaRange(cast(void*)aa));
+    return () @trusted { return Result(_aaRange(cast(void*)aa)); } ();
```

Comment:
The constructor of `Result` should be safe, so you could try to move the lamda inside.